sotodlib/coords/pmat.py

```diff
@@ -123,23 +123,26 @@ class P:
 
     """
     def __init__(self, sight=None, fp=None, geom=None, comps='T',
-                 cuts=None, threads=None, det_weights=None, interpol=None):
+                 cuts=None, threads=None, det_weights=None, interpol=None, hp_geom=None):
         self.sight = sight
         self.fp = fp
-        self.geom = wrap_geom(geom)
+        if geom is not None:
+            self.geom = wrap_geom(geom)
         self.comps = comps
         self.cuts = cuts
         self.threads = threads
         self.active_tiles = None
         self.det_weights = det_weights
         self.interpol = interpol
+        self.hp_geom = hp_geom
+        self.hp_tiled = self.hp_geom is not None and self.hp_geom.nside_tile is not None
```

Comment:
You can just reuse self.tiled, right?

It is a bug (which I have fixed in a brack somewhere) that this constructor doesn't set `self.tiled = None`.  So I suggest you do that, and than update it to self.hp_geom.nside_tile if conditions support that.

There are a lot of places where you need to do things differently depending on rectpix vs. healpix.  So consider inventing enumerator `self.pix_scheme`, with values none, rectpix, healpix, and choose branches based on that rather than depending on what things are None.

Reply:
Thanks for the suggestion, I have implemented it.
`self.tiled` is implemented here as a property toward the bottom of the file, so it can't be directly initialized to None. That should now work nicely with healpix as well though, so I have merged them (and geom / hp_geom)